konduit-serving-test/src/test/java/ai/konduit/serving/pipeline/steps/TransformProcessStepTest.java

```diff
@@ -28,17 +27,16 @@ public void testTransformProcessPipe() throws Exception {
                 .step("foo", tp, outputSchema)
                 .step("bar", tp, outputSchema);
 
-        assert tpps.getInputNames().contains("foo");
-        assert tpps.getInputNames().contains("bar");
+        assertTrue(tpps.getInputNames().contains("foo"));
+        assertTrue(tpps.getInputNames().contains("bar"));
 
-        assert tpps.getOutputNames().contains("foo");
-        assert tpps.getOutputNames().contains("bar");
+        assertTrue(tpps.getOutputNames().contains("foo"));
+        assertTrue(tpps.getOutputNames().contains("bar"));
 
         TransformProcessStep tpps2 = new TransformProcessStep(tp, outputSchema);
 
-        assert tpps2.getTransformProcesses().get("default") == tp;
-
-        assert Arrays.equals(tpps2.getInputSchemas().get("default"), SchemaTypeUtils.typesForSchema(inputSchema));
+        assertSame(tpps2.getTransformProcesses().get("default"), tp);
```

Comment:
do we want assertSame, or assertEquals here?
assertSame -> must be exact same object (and hence by definition also equal)
assertEquals -> can be different object, but still has to be equal

Reply:
I think assertSame would make more sense since what we're trying to check is that if the constructor and the builder method are assigning the same object to the different input names for steps.